tests/adapter/read.ts

```diff
@@ -0,0 +1,58 @@
+import { expect } from "chai";
+import nock from "nock";
+
+import ImageKitAdapter from "../../src";
+import ImageKitAdapterError from "../../utils/errors";
+import { testConfig } from "../fixtures";
+
+const imagekitAdapter = new ImageKitAdapter(testConfig());
+
+describe("read", function () {
+  it("Should read the image", async function () {
+    nock("https://ik.imagekit.io")
+      .get("/test/default-image.jpg")
+      .reply(200, "imagebuffer");
+
+    const options = {
+      path: "https://ik.imagekit.io/test/default-image.jpg",
+    };
+
+    const buffer = await imagekitAdapter.read(options);
+
+    nock.restore();
+
+    expect(buffer.toString()).to.equal("imagebuffer");
+  });
+
+  it("Should return an error for the missing options parameter", async function () {
+    try {
+      await imagekitAdapter.read();
+    } catch (err) {
+      expect(err).to.be.an.instanceOf(ImageKitAdapterError);
+      expect(err.statusCode).to.be.equal(400);
+      expect(err.message).to.be.equal(
+        "Failed to read image from path 'undefined'"
+      );
+    }
+  });
+
+  it("Should return an error for a non-existent file", async function () {
```

Comment:
Can we add a test for private file?

Reply:
Done. Added for signed URL setting also.